### ISSUE TITLE
[BR]: ignorecommands on FreeBSD

### ISSUE THREAD

**Issue by ohreally** [fail2ban/fail2ban#3005]
Hi,

The following line should be added to _config/paths-freebsd.conf_:

`ignorecommands_dir = /usr/local/etc/fail2ban/filter.d/ignorecommands`

Cheers,
  Rob

**Comment by sebres** [comment]
Are the configs really installed in `/usr/local/etc/fail2ban` on freebsd? and was it manually or from distro installed?

BTW, I guess one could simply rewrite this using relative path (must try):
https://github.com/fail2ban/fail2ban/blob/6893d5a8b78fbdf263df8a233b3b9ac343c7a02c/config/jail.conf#L349
like:
```diff
- ignorecommand = %(ignorecommands_dir)s/apache-fakegooglebot <ip>
+ ignorecommand = filter.d/ignorecommands/apache-fakegooglebot <ip>
```
(so we would not need this variable at all)

**Comment by sebres** [comment]
Oh, since it is a command, I don't think it would work with relative path.
But we could provide some internal interpolation like `%(fail2ban_config)s` (similar to `%(fail2ban_version)s` etc).

**Comment by ohreally** [comment]
On FreeBSD, all non-OS applications are installed in `/usr/local`; all non-OS configs go in `/usr/local/etc`.
It was installed from the ports collection (which would be 'distro' in Linux-speak).

**Comment by chtheis** [comment]
Rob is right, FreeBSD uses different path prefixes for external
programs. But: it is possible to install such programs with any prefix
and the ports' Makefile does some sed to correct pathes. E.g. one could
install ports into /home/ instead of /usr/local. Isn't that the use case
for jails?

So we have different options:
1) Let Makefile correct the path in pathes-common
2) Let Makefile add the path to pathes-freebsd
3) Put a line into pathes-freebsd and let Makefile correct it

Any FreeBSD users here who can tell the canonical way to do it?


--
Christoph

**Comment by ohreally** [comment]
I think in those cases you would use `paths-overrides.local`.
Apache and Nginx can also be configured to log elsewhere (as many sysadmins probably do), but `paths-*.conf` still list some defaults.

And otherwise, I guess it would be a job for the port maintainer: https://www.freshports.org/security/py-fail2ban/
I imagine the ports collection has some built-in functionality in it's Makefiles to replace `/usr/local` with something user-defined.

**Comment by chtheis** [comment]
On 16.04.2021 10:33, Robert LA LAU wrote:
> And otherwise, I guess it would be a job for the port maintainer:

Which would be me :)

> https://www.freshports.org/security/py-fail2ban/
> <https://www.freshports.org/security/py-fail2ban/>
> I imagine the ports collection has some built-in functionality in it's
> Makefiles to replace |/usr/local| with something user-defined.

In the Makefile you'd use $ETCDIR, $LOCALBASE, and the like. But the
.conf files are not subject to make. In other cases I use sed to change
pathes to the correct ones, but in this case we try to have correct path
in the file which then we have to change again.

I don't like to have overrides here, a port should install with correct
pathes.


--
Christoph

**Comment by ohreally** [comment]
Okay then.
Well, in that case: fail2ban gets it configuration directory path somewhere. I'd say that would be the spot to set a configuration variable `%(config_dir)s`, which could then be used throughout the configuration, and would always be portable.

**Comment by sebres** [comment]
Fixed now in a2431158f670867e65534674b640cd0672a85e49 for master (v.1.0) using new interpolation variable `%(fail2ban_confpath)s` (automatically substituted from config reader path, e.g. from `-c` parameter of command-line). 

New `jail.conf` entry:
```ini
ignorecommand = %(fail2ban_confpath)s/filter.d/ignorecommands/apache-fakegooglebot <ip>
```
makes `ignorecommands_dir` from `paths-common.conf` completely obsolete, thus removed from there.
If it is still needed for some reason, it can be rewritten like here:
```diff
- ignorecommands_dir = /etc/fail2ban/filter.d/ignorecommands
+ ignorecommands_dir = <fail2ban_confpath>/filter.d/ignorecommands
```

If fix is required for v.0.11/0.10 the revision could be simply cherry-picked.
